src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java

```diff
@@ -273,6 +274,23 @@ public void exit(AbruptExitException exception) {
     }
     this.buildResultListener = new BuildResultListener();
     this.eventBus.register(this.buildResultListener);
+
+    ImmutableMap.Builder<String, Path> wellKnownRoots = ImmutableMap.builder();
+    // This is necessary because some tests don't have a workspace set.
+    putIfValueNotNull(wellKnownRoots, "workspace", directories.getWorkspace());
```

Comment:
The helper isn't used anywhere else, so just inline the logic here?

Reply:
Moved all logic into a static factory function for `CommandLinePathFactory`.